lib/stdlib/test/dets_SUITE.erl

```diff
@@ -376,9 +376,9 @@ dirty_mark(Config) when is_list(Config) ->
     {ok, Node} = test_server:start_node(dets_dirty_mark,
                                         slave,
                                         [{linked, false},
-                                         {args, "-pa " ++
+                                         {args, ["-pa",
                                               filename:dirname
-						(code:which(?MODULE))}]),
+						(code:which(?MODULE))]}]),
```

Comment:
Why isn't this call replaced with a call to `peer`? Isn't the `peer` module powerful enough?

Reply:
I (incorrectly) assumed that `test_server:start_node` is used by OTP test suites to channel `start_node` calls through some Test Server that runs somewhere else, not on the machine where test cases are actually executed. 

From my reading of `test_server:start_node`, it calls group leader (which may reside anywhere in the connected cluster) to start a new ERTS instance. So technically it is possible that test cases are running on `runner@host_a`, but group leader is running on node `test_server@host_b`. In this case new nodes will be started on `host_b`. In my change I retained this behaviour.

If OTP test cases are always running in the same host with `test_server`, it significantly simplifies everything! And lets me to simply use `peer:start_link` instead of `test_server`. Would you be ok with this approach? If yes, let me just replace all `test_server:start_node` with `peer:start_link`, it will be so much easier to debug.

Reply:
> If OTP test cases are always running in the same host with test_server, it significantly simplifies everything! 

Yes, that should be the case as far as I know. Some of the code you saw probably remains from the days when we supported VxWorks. Only one Erlang node could run on each VxWorks host, so all slave nodes were remote.

However, when discussing this PR with @garazdawi, he pointed out that `test_server:start_node()` sets up locations for crash dumps so that they can be copied to the test case log. Perhaps we should invent a new module `ct_peer` that is a thin wrapper on top of `peer` that can handle such details. Your suggested `?CT_PEER` macro could call `ct_peer`. (I like your `?CT_PEER` macro because it invents a node name that is unlikely to collide with other node names; we have had a lot of problems with nodes remaining after a failed test case and other test cases that used the same name also failed.)